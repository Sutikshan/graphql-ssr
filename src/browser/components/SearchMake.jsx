import React from "react";
import PropTypes from "prop-types";
import { graphql } from "react-apollo";
import { makeQuery } from "./queries";

export function SearchMake({ data, onMakeSelect, makeId }) {
  let { makes, loading } = data;
  if (loading) {
    makes = [];
  }

  return (
    <div>
      <select onChange={onMakeSelect} value={makeId}>
        <option>Select Car Make</option>
        {makes.map(make => {
          return (
            <option key={make.id} value={make.id}>
              {make.name}
            </option>
          );
        })}
      </select>
    </div>
  );
}

SearchMake.propTypes = {
  data: PropTypes.shape({
    makes: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string
      })
    )
  }),
  onMakeSelect: PropTypes.func.isRequired,
  makeId: PropTypes.number
};

export default graphql(makeQuery)(SearchMake);
