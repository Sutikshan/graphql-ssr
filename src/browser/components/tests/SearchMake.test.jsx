import React from "react";
import { shallow } from "enzyme";
import { SearchMake } from "../SearchMake";

it("should render a SearchMake component correctly", () => {
  const data = {
    makes: [{ id: 1, Name: "Chevy" }]
  };
  const wrapper = shallow(<SearchMake data={data} onMakeSelect={() => {}} />);

  expect(wrapper).toMatchSnapshot();
});
