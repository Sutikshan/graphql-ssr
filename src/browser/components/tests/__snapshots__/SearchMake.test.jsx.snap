// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a SearchMake component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchMake
    data={
      Object {
        "makes": Array [
          Object {
            "Name": "Chevy",
            "id": 1,
          },
        ],
      }
    }
    onMakeSelect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[Function]}
        value={undefined}
      >
        <option>
          Select Car Make
        </option>
        <option
          value={1}
        />
      </select>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option>
            Select Car Make
          </option>,
          Array [
            <option
              value={1}
            />,
          ],
        ],
        "onChange": [Function],
        "value": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select Car Make",
          },
          "ref": null,
          "rendered": "Select Car Make",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "value": 1,
          },
          "ref": null,
          "rendered": null,
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[Function]}
          value={undefined}
        >
          <option>
            Select Car Make
          </option>
          <option
            value={1}
          />
        </select>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option>
              Select Car Make
            </option>,
            Array [
              <option
                value={1}
              />,
            ],
          ],
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select Car Make",
            },
            "ref": null,
            "rendered": "Select Car Make",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
